aether-ui/aether/ui/assets/apps/pipeline/Pipeline.jsx

```diff
@@ -111,18 +115,52 @@ class Pipeline extends Component {
     }
   }
 
+  linksCallBack () {
+    this.checkUnsavedContract(() => {
+      this.props.history.push('/')
+      this.props.clearSelection()
+    })
+  }
+
+  checkUnsavedContract (cb) {
+    if (this.state.newContract) {
+      this.setState({
+        showCancelModal: true,
+        onContractSavedCallback: cb
+      })
+    } else {
+      this.setState({
+        showCancelModal: false
+      })
+      if (cb) {
+        cb()
+      }
+    }
+  }
+
   render () {
     const { pipeline } = this.props
     if (!pipeline) {
       return '' // still loading data
     }
 
+    const fullscreenDiv = (
+      <div
+        className='btn-icon fullscreen-toggle'
+        onClick={() => { this.setState({ fullscreen: !this.state.fullscreen }) }}>
+        { this.state.fullscreen
+          ? <FormattedMessage id='pipeline.navbar.shrink' defaultMessage='fullscreen off' />
+          : <FormattedMessage id='pipeline.navbar.fullscreen' defaultMessage='fullscreen on' />
+        }
+      </div>
+    )
+
     return (
       <div className={`pipelines-container show-pipeline pipeline--${this.state.view}`}>
         { this.props.error && <ModalError error={this.props.error} /> }
-        <NavBar showBreadcrumb>
+        <NavBar showBreadcrumb cb={this.linksCallBack.bind(this)}>
```

Comment:
Please put all the bind calls into the constructor, e.g.
```
this.linksCallBack = this.linksCallBack.bind(this);
```

It makes the JSX more readable and removes duplication.